### ISSUE TITLE
db_redis location schema

### ISSUE THREAD

**Issue by juansalvatella** [kamailio/kamailio#1917]
### Description

Configured the location module to load the expires column as a datetime instead of an integer but db_redis ignored the parameter and loaded the data with the default schema. As of today, the default schema has last_modified and expires columns as integers but stores them as timestamps.

```
id/int,ruid/string,username/string,domain/string,contact/string,received/string,path/string,expires/int,q/double,callid/string,cseq/int,last_modified/int,flags/int,cflags/int,user_agent/string,socket/string,methods/int,instance/string,reg_id/int,server_id/int,connection_id/int,keepalive/int,partition/int,
9
```

#### Reproduction

Use db_redis as backend and use the location module.

While handling INVITE:

```
if (!lookup("location")) {
	sl_send_reply("404", "User Not Found");
	return;
}
```

While handling REGISTER:

```
if (!save("location"))
      sl_reply_error();
```

#### Log Messages

```
8(34) ERROR: <core> [db_ut.c:92]: db_str2int(): Unexpected characters: [-04-01 07:17:14]
 8(34) ERROR: <core> [db_val.c:76]: db_str2val(): error while converting integer value from string
 8(34) ERROR: db_redis [redis_dbase.c:1062]: db_redis_convert_row(): Failed to convert redis column 'expires' to db value
 8(34) ERROR: db_redis [redis_dbase.c:1231]: db_redis_perform_query(): Failed to convert redis reply for row
 8(34) ERROR: db_redis [redis_dbase.c:1243]: db_redis_perform_query(): failed to perform the query
 8(34) ERROR: db_redis [redis_dbase.c:1829]: db_redis_query(): failed to do the query
 8(34) ERROR: usrloc [udomain.c:641]: db_load_urecord(): db_query failed
```

#### SIP Traffic

```
(paste your sip traffic here)
```

### Possible Solutions

I patched the default schema changing expires and last_modified from int to timestamp.

```
id/int,ruid/string,username/string,domain/string,contact/string,received/string,path/string,expires/timestamp,q/double,callid/string,cseq/int,last_modified/timestamp,flags/int,cflags/int,user_agent/string,socket/string,methods/int,instance/string,reg_id/int,server_id/int,connection_id/int,keepalive/int,partition/int,
9
```

### Additional Information

  * **Kamailio Version** - output of `kamailio -v`

```
version: kamailio 5.2.2 (x86_64/linux) 
flags: STATS: Off, USE_TCP, USE_TLS, USE_SCTP, TLS_HOOKS, USE_RAW_SOCKS, DISABLE_NAGLE, USE_MCAST, DNS_IP_HACK, SHM_MEM, SHM_MMAP, PKG_MALLOC, Q_MALLOC, F_MALLOC, TLSF_MALLOC, DBG_SR_MEMORY, USE_FUTEX, FAST_LOCK-ADAPTIVE_WAIT, USE_DNS_CACHE, USE_DNS_FAILOVER, USE_NAPTR, USE_DST_BLACKLIST, HAVE_RESOLV_RES
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144 MAX_URI_SIZE 1024, BUF_SIZE 65535, DEFAULT PKG_SIZE 8MB
poll method support: poll, epoll_lt, epoll_et, sigio_rt, select.
id: unknown 
compiled with gcc 6.3.0
```

* **Operating System**:

```
Linux gke-production-cluster-default-pool-fb621121-qz3r 4.14.65+ #1 SMP Thu Oct 25 10:42:50 PDT 2018 x86_64 GNU/Linux
```

**Comment by miconda** [comment]
You mentioned a patch, so if you have a commit that you want to be reviewed and eventually merged as a fix, then make a pull request with it.

**Comment by linuxmaniac** [comment]
I think the root cause is this that c9f2aa71b2e0d4a4f71b3da92f97306e86f93eea was manually updated and lost at 5e0440aa27154c263fc883f3a7cb9680805af6c3

#1919 will restore the intended changes

**Comment by linuxmaniac** [comment]
backported fixes to 5.2 branch too
